geomesa-features/geomesa-feature-kryo/src/main/scala/org/locationtech/geomesa/features/kryo/KryoBufferSimpleFeature.scala

```diff
@@ -126,22 +126,30 @@ class KryoBufferSimpleFeature(sft: SimpleFeatureType,
         case _ => -1
       }
     }
+
+    val shouldReserialize = indices.contains(-1)
+    val mutableOffsetsAndLength = Array.ofDim[(Int,Int)](indices.length)
```

Comment:
we could move this down inside the if on line 135 (but before the function definition) so that we don't create it in the reserialize case